libnmstate/metadata.py

```diff
@@ -65,6 +71,10 @@ def generate_ifaces_metadata(desired_state, current_state):
         get_slaves_func=linux_bridge.get_slaves_from_state,
         set_metadata_func=linux_bridge.set_bridge_ports_metadata
     )
+    # _generate_dns_metadata() should be executed before
```

Comment:
Not just that, in `_generate_dns_metadata()` we include a interface for DNS with name only without route metadata.

Reply:
That does not contradict anything else. One adds dns metadata and the other route metadata. if the interface exists in current but not in desired, it will be added to desired.